### PR TITLE
Fixes #8886 feat(nimbus):change adds tailwind css support to the project

### LINKED ISSUE #8886: Static assets
Set up static assets (Tailwind, Alpine, and Font Awesome)

1. Install the necessary packages and libraries
1. Add the assets to the app's static directory
1. Ensure the assets are properly loaded in the templates

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-3532)

### PR DESCRIPTION
**Pull request by Avi-88**
Because

- the new 'history' page implementation requires tailwind css as a dependency

This commit

- adds tailwind css in to the project by using the django-tailwind package

### CODE REVIEW

**Comment by Avi-88** [comment]
> Hey @Avi-88, taking a look now.

Yes, thank you 🙌

**Comment by Avi-88** [comment]
> @Avi-88 as per our discussion, have you tried using different version?

Yes, I did try it with an older version but the requirements for that don't change. Every version of the package requires python 3.8 or newer with django >= 3.2

**Comment by jrbenny35** [comment]
Hey @Avi-88, so I tried your branch locally and experimenter won't start. I did a ```make refresh``` followed by a ```make up_prod``` and I am getting some errors.

```
experimenter-experimenter-1  | Traceback (most recent call last):
experimenter-experimenter-1  |   File "/experimenter/manage.py", line 10, in <module>
experimenter-experimenter-1  |     execute_from_command_line(sys.argv)
experimenter-experimenter-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
experimenter-experimenter-1  |     utility.execute()
experimenter-experimenter-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 395, in execute
experimenter-experimenter-1  |     django.setup()
experimenter-experimenter-1  |   File "/usr/local/lib/python3.11/site-packages/django/__init__.py", line 24, in setup
experimenter-experimenter-1  |     apps.populate(settings.INSTALLED_APPS)
experimenter-experimenter-1  |   File "/usr/local/lib/python3.11/site-packages/django/apps/registry.py", line 91, in populate
experimenter-experimenter-1  |     app_config = AppConfig.create(entry)
experimenter-experimenter-1  |                  ^^^^^^^^^^^^^^^^^^^^^^^
experimenter-experimenter-1  |   File "/usr/local/lib/python3.11/site-packages/django/apps/config.py", line 224, in create
experimenter-experimenter-1  |     import_module(entry)
experimenter-experimenter-1  |   File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
experimenter-experimenter-1  |     return _bootstrap._gcd_import(name[level:], package, level)
experimenter-experimenter-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
experimenter-experimenter-1  |   File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
experimenter-experimenter-1  |   File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
experimenter-experimenter-1  |   File "<frozen importlib._bootstrap>", line 1142, in _find_and_load_unlocked
experimenter-experimenter-1  | ModuleNotFoundError: No module named 'theme'
experimenter-experimenter-1 exited with code 1
```

**Comment by Avi-88** [comment]
> Hey @Avi-88, so I tried your branch locally and experimenter won't start. I did a `make refresh` followed by a `make up_prod` and I am getting some errors.
> 
> ```
> experimenter-experimenter-1  | Traceback (most recent call last):
> experimenter-experimenter-1  |   File "/experimenter/manage.py", line 10, in <module>
> experimenter-experimenter-1  |     execute_from_command_line(sys.argv)
> experimenter-experimenter-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
> experimenter-experimenter-1  |     utility.execute()
> experimenter-experimenter-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 395, in execute
> experimenter-experimenter-1  |     django.setup()
> experimenter-experimenter-1  |   File "/usr/local/lib/python3.11/site-packages/django/__init__.py", line 24, in setup
> experimenter-experimenter-1  |     apps.populate(settings.INSTALLED_APPS)
> experimenter-experimenter-1  |   File "/usr/local/lib/python3.11/site-packages/django/apps/registry.py", line 91, in populate
> experimenter-experimenter-1  |     app_config = AppConfig.create(entry)
> experimenter-experimenter-1  |                  ^^^^^^^^^^^^^^^^^^^^^^^
> experimenter-experimenter-1  |   File "/usr/local/lib/python3.11/site-packages/django/apps/config.py", line 224, in create
> experimenter-experimenter-1  |     import_module(entry)
> experimenter-experimenter-1  |   File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
> experimenter-experimenter-1  |     return _bootstrap._gcd_import(name[level:], package, level)
> experimenter-experimenter-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> experimenter-experimenter-1  |   File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
> experimenter-experimenter-1  |   File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
> experimenter-experimenter-1  |   File "<frozen importlib._bootstrap>", line 1142, in _find_and_load_unlocked
> experimenter-experimenter-1  | ModuleNotFoundError: No module named 'theme'
> experimenter-experimenter-1 exited with code 1
> ```

Oh, I cant seem to recreate this but can you try running this command inside the project `python manage.py tailwind install` so that it installs all the packages required for the theme app. I'm guessing this is what is causing the tests to fail as well. // @jrbenny35

**Comment by jrbenny35** [comment]
> 
> Oh, I cant seem to recreate this but can you try running this command inside the project `python manage.py tailwind install` so that it installs all the packages required for the theme app. I'm guessing this is what is causing the tests to fail as well. // @jrbenny35

Try starting over with a ```make refresh```. That could be the problem as the tests start fresh each time.

**Comment by Avi-88** [comment]
> > Oh, I cant seem to recreate this but can you try running this command inside the project `python manage.py tailwind install` so that it installs all the packages required for the theme app. I'm guessing this is what is causing the tests to fail as well. // @jrbenny35
> 
> Try starting over with a `make refresh`. That could be the problem as the tests start fresh each time.

I did. I ran `make refresh` as well as `make check` and both didn't throw any errors.

File: experimenter/experimenter/theme/static_src/package.json
Comment (brennie) [inline experimenter/experimenter/theme/static_src/package.json:9]: We shouldn't need `cross-env`. We only support building our container on Linux and macOS, which support the `env` command.

File: experimenter/experimenter/theme/static_src/package.json
Comment (brennie) [inline experimenter/experimenter/theme/static_src/package.json:1]: Can you add `private: true` to this manifest?

File: experimenter/experimenter/theme/static_src/package.json
Comment (brennie) [inline experimenter/experimenter/theme/static_src/package.json:2]: Can we rename this to e.g., `experimenter-theme` ?

File: experimenter/experimenter/theme/static_src/package.json
Comment (brennie) [inline experimenter/experimenter/theme/static_src/package.json:8]: Likewise, we don't need `rimraf` on Mac and Linux.

File: experimenter/experimenter/theme/static_src/package.json
Comment (brennie) [inline experimenter/experimenter/theme/static_src/package.json:6]: I don't see the point of aliasing start to dev. I'd remove this.

File: experimenter/experimenter/theme/static_src/package.json
Comment (brennie) [inline experimenter/experimenter/theme/static_src/package.json:11]: scripts are able to find binaries that are installed to `node_modules/.bin`, so you should be able to do:
```
"tailwindcss": "tailwindcss"
```

File: experimenter/experimenter/theme/static_src/package.json
Comment (brennie) [inline experimenter/experimenter/theme/static_src/package.json:15]: This should be MPL-2.0.

File: experimenter/experimenter/theme/static_src/package.json
Comment (brennie) [inline experimenter/experimenter/theme/static_src/package.json:14]: Our other package.jsons either omit this or have "Mozilla" as the value. I prefer the latter, but either is fine.

**Review by brennie — CHANGES_REQUESTED** [review]

File: experimenter/experimenter/settings.py
Comment (yashikakhurana) [inline experimenter/experimenter/settings.py:104]: you can remove this line

**Review by yashikakhurana — CHANGES_REQUESTED** [review]
thank you @brennie for the review, completely +100 with your thoughts, other than that @Avi-88 great job finding and fixing that, one minor change other than that, I am so excited to see it is coming together 👏

File: docker-compose.yml
Comment (eliserichards) [inline docker-compose.yml:26]: I'd love another pair of eyes on this since I'm not very familiar with the docker compose file

File: experimenter/experimenter/settings.py
Comment (eliserichards) [inline experimenter/experimenter/settings.py:106]: 🔥🔥🔥

**Review by eliserichards — COMMENTED** [review]
This is awesome 👏 I want at least one more pair of eyes on this before we land, but this is looking great.

File: experimenter/Dockerfile
Comment (mikewilli) [inline experimenter/Dockerfile:66]: Here and line 88 should also be able to use `--cwd` parameter.

File: experimenter/experimenter/theme/static_src/package.json
Comment (brennie) [inline experimenter/experimenter/theme/static_src/package.json:26]: Missing newline at end of file

**Review by brennie — CHANGES_REQUESTED** [review]

File: experimenter/experimenter/theme/static_src/package.json
Comment (Avi-88) [reply experimenter/experimenter/theme/static_src/package.json]: I've configured my vs code workspace to do this but it seems that isn't working

File: docker-compose.yml
Comment (Avi-88) [reply docker-compose.yml]: This I changed from `&&` to `&` as the second one didn't execute because the `runserver` doesn't terminate. So to run both  commands I used the background operator `&` , [here's](https://stackoverflow.com/questions/3683910/executing-shell-command-in-background-from-script) a reference

Comment (mikewilli) [reply docker-compose.yml]: Ah yes good point, thanks for clarifying!

Does it make more sense to run the frontend in a separate container? Feels odd to background one process so you can run another in the docker context, but maybe this is normal.

If you keep it all in one container, consider putting each command on its own line for readability purposes.

**Review by brennie — APPROVED** [review]
Let's goooo!

File: docker-compose.yml
Comment (brennie) [inline docker-compose.yml:26]: Actually, one final nit:

Can we rewrite this with the yarn cmd running in the background perhaps?

Comment (brennie) [reply docker-compose.yml:26]:
```
    command: bash -c "python bin/setup_kinto.py;/experimenter/bin/wait-for-it.sh db:5432 --  yarn --cwd /experimenter/experimenter/theme/static_src run dev &; python /experimenter/manage.py runserver 0:7001"
```

**Review by brennie — CHANGES_REQUESTED** [review]
I'm looking at the django-tailwind docs and their example uses a separate container:

https://github.com/timonweb/django-tailwind/blob/master/example/docker-compose.yml#L17

I think thats the way we should go.

File: docker-compose.yml
Comment (brennie) [reply docker-compose.yml:26]: see my other comment

File: experimenter/experimenter/templates/changelog/overview.html
Comment (brennie) [inline experimenter/experimenter/templates/changelog/overview.html:16]: I know fontawesome recommends using `<i>` for icons, but it's not great practice for semantic HTML. We should use `<span>`s here.

**Review by brennie — CHANGES_REQUESTED** [review]
One final nit, then this is good to land.

**Review by brennie — APPROVED** [review]

**Pull request merged**
